subxt/src/blocks/extrinsic_types.rs

```diff
@@ -558,6 +607,116 @@ impl<T: Config> ExtrinsicEvents<T> {
     }
 }
 
+/// The signed extensions of an extrinsic.
+#[derive(Debug, Clone)]
+pub struct ExtrinsicSignedExtensions<'a> {
+    bytes: &'a [u8],
+    metadata: &'a Metadata,
+}
+
+impl<'a> ExtrinsicSignedExtensions<'a> {
+    /// Returns an iterator over each of the signed extension details of the extrinsic.
+    /// If the decoding of any signed extension fails, an error item is yielded and the iterator stops.
+    pub fn iter(&self) -> impl Iterator<Item = Result<ExtrinsicSignedExtension<'a>, Error>> {
+        let signed_extension_types = self.metadata.extrinsic().signed_extensions();
+        let num_signed_extensions = signed_extension_types.len();
+        let bytes = self.bytes;
+        let metadata = self.metadata;
+        let mut index = 0;
+        let mut byte_start_idx = 0;
+
+        std::iter::from_fn(move || {
+            if index == num_signed_extensions {
+                return None;
+            }
+
+            let extension = &signed_extension_types[index];
+            let ty_id = extension.extra_ty();
+            let cursor = &mut &bytes[byte_start_idx..];
+            if let Err(err) = scale_decode::visitor::decode_with_visitor(
+                cursor,
+                ty_id,
+                metadata.types(),
+                scale_decode::visitor::IgnoreVisitor,
+            )
+            .map_err(|e| Error::Decode(e.into()))
+            {
+                index = num_signed_extensions; // (such that None is returned in next iteration)
+                return Some(Err(err));
+            }
+            let byte_end_idx = bytes.len() - cursor.len();
+            let bytes = &bytes[byte_start_idx..byte_end_idx];
+            byte_start_idx = byte_end_idx;
+            index += 1;
+            Some(Ok(ExtrinsicSignedExtension {
+                bytes,
+                ty_id,
+                identifier: extension.identifier(),
+                metadata,
+            }))
+        })
+    }
+
+    /// The tip of an extrinsic, extracted from the `ChargeTransactionPayment` or
+    /// `ChargeAssetTxPayment` signed extension, depending on which is present.
+    pub fn tip(&self) -> Option<u128> {
```

Comment:
Might be worth extending these functions to also fetch:
- transaction version (CheckTxVersion)
- spec version (CheckSpecVersion)
- tx genesis (CheckGenesis)
- tx weigth (CheckWeight)

Totally up to you tho

Reply:
Personally I'd just start with a couple of obvious ones that people def want to begin with, since they all rely on specific signed extensions existing to work :)

Reply:
yeah, let's do in a follow-up PR if needed